homeassistant/components/github/coordinator.py

```diff
@@ -1,44 +1,92 @@
-"""Custom data update coordinators for the GitHub integration."""
+"""Custom data update coordinator for the GitHub integration."""
 from __future__ import annotations
 
-from typing import Literal, TypedDict
+from typing import Any
 
 from aiogithubapi import (
     GitHubAPI,
-    GitHubCommitModel,
     GitHubConnectionException,
     GitHubException,
-    GitHubNotModifiedException,
     GitHubRatelimitException,
-    GitHubReleaseModel,
-    GitHubRepositoryModel,
     GitHubResponseModel,
 )
 
-from homeassistant.config_entries import ConfigEntry
-from homeassistant.core import HomeAssistant, T
+from homeassistant.core import HomeAssistant
 from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
 
-from .const import DEFAULT_UPDATE_INTERVAL, DOMAIN, LOGGER, IssuesPulls
-
-CoordinatorKeyType = Literal["information", "release", "issue", "commit"]
-
-
-class GitHubBaseDataUpdateCoordinator(DataUpdateCoordinator[T]):
-    """Base class for GitHub data update coordinators."""
+from .const import DEFAULT_UPDATE_INTERVAL, DOMAIN, LOGGER
+
+GRAPHQL_REPOSITORY_QUERY = """
+query ($owner: String!, $repository: String!) {
+  rateLimit {
+    cost
+    remaining
+  }
```

Comment:
This is not directly used by the integration, but it's useful to have it in here for when we want to change/expand this query in the future to make sure we don't add anything that will use too much of the limit. (currently the cost is 1 for this query)